src/mcode-pilot/utils/filterTreatmentData.js

```diff
@@ -62,29 +41,29 @@ function generateTreatmentData(similarPatients, treatments, includedTreatments)
     if (similarPatients.length === 0) return [];
     let treatmentData = [];
     treatments.forEach(treatment => {
-        const filteredPatients = similarPatients.filter(patient => isSame(patient.treatments, treatment));
+        const filteredPatients = similarPatients.filter(patient => isSame(patient.treatments.map((e) => { return typeof e === 'object' ?e.code : e; }), treatment.map(treat => { return treat.code; })));
         let displayName =
             _.isArray(treatment)
                 ? (treatment === includedTreatments)
-                    ? treatment.map(name => TREATMENT_NAMES[name]).join(' & ')
+                    ? treatment.map(name => name.displayName).join(' & ')
```

Comment:
Seeing how name is no longer a sting/name can we change the variable name to something more appropriate